Back out "Make adding buffers more like adding parameters (#104069)"

atalman opened the pull request:
Summary:
D47537831 is breaking pyper tests: https://fb.workplace.com/groups/802176577445480/posts/1018902842439518/

with `TypeError: register_buffer() takes 3 positional arguments but 4 were given`

Original commit changeset: d4b4069fbd38

Original Phabricator Diff: D47537831

Test Plan:
```
buck2 run //caffe2/torch/fb/training_toolkit/integration_tests/training_lifecycle/cogwheel_tests/pyper_release_v2:cogwheel_smallworld_inline_cvr_infer_pyper_pyper__canary_offline_training-launcher -- --run-harness-in-tupperware --build-fbpkg ads_dper3 --build-fbpkg training_platform
```

Reviewed By: atalman

Differential Revision: D47600140



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @anijain2305

albanD commented:
for future reference, this is not a problem from the PR itself but bad subclassing.

albanD reviewed (commented):
Revert is fine but CLA needs signing.

atalman committed:
Back out "Make adding buffers more like adding parameters (#104069)" (#105581)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/105581

D47537831 is breaking pyper tests: https://fb.workplace.com/groups/802176577445480/posts/1018902842439518/

with `TypeError: register_buffer() takes 3 positional arguments but 4 were given`

Original commit changeset: d4b4069fbd38

Original Phabricator Diff: D47537831

Test Plan:
```
buck2 run //caffe2/torch/fb/training_toolkit/integration_tests/training_lifecycle/cogwheel_tests/pyper_release_v2:cogwheel_smallworld_inline_cvr_infer_pyper_pyper__canary_offline_training-launcher -- --run-harness-in-tupperware --build-fbpkg ads_dper3 --build-fbpkg training_platform
```

Reviewed By: mikaylagawarecki

Differential Revision: D47600140

fbshipit-source-id: 224a1bb94001ed409262c06f84476d879c73460b

kit1980 commented:
/easycla

atalman commented:
/easycla

ekamiti commented:
Unfortunate that this change broke something. @mikaylagawarecki @albanD Is there anything I need to do for this as the author for https://github.com/pytorch/pytorch/pull/104069? I am assuming that these pyper tests are internal fb tests that I don't have access to?

mikaylagawarecki commented:
@ekamiti yep the failure is not the fault of your PR but rather some issues with overriding. There shouldn't be an action item for you here!

wanchaol commented:
@atalman I think this revert broke the CI, i.e. https://ossci-raw-job-status.s3.amazonaws.com/log/15189813599

The reason is that the revert accidentally delete some necessary logic to handle tensor subclass that should be preserved (there were some merge conflicts between the original change in https://github.com/pytorch/pytorch/pull/104069 and https://github.com/pytorch/pytorch/pull/105308 and this revert diff does not resolve correctly). I am fixing the issue in https://github.com/pytorch/pytorch/pull/105797
